src/qt/bitcoingui.cpp

```diff
@@ -889,16 +889,18 @@ void BitcoinGUI::createIconMenu(QMenu *pmenu)
     // Configuration of the tray icon (or dock icon) icon menu
     pmenu->addAction(toggleHideAction);
     pmenu->addSeparator();
-    pmenu->addAction(sendCoinsMenuAction);
-    pmenu->addAction(coinJoinCoinsMenuAction);
-    pmenu->addAction(receiveCoinsMenuAction);
-    pmenu->addSeparator();
-    pmenu->addAction(signMessageAction);
-    pmenu->addAction(verifyMessageAction);
-    pmenu->addSeparator();
+    if (enableWallet) {
+        pmenu->addAction(sendCoinsMenuAction);
+        pmenu->addAction(coinJoinCoinsMenuAction);
+        pmenu->addAction(receiveCoinsMenuAction);
+        pmenu->addSeparator();
+        pmenu->addAction(signMessageAction);
+        pmenu->addAction(verifyMessageAction);
+        pmenu->addSeparator();
+        pmenu->addAction(openRPCConsoleAction);
+    }
```

Comment:
14383: should also move `openRepairAction` and `showBackupsAction` here I think

Reply:
What about the item order? Should I keep it or group everything here?

Reply:
I would say keep the previous ordering, just move it all into the if block

Reply:
🤦 
Asking before pushing again:

Option 1) all into the same if block
```
void BitcoinGUI::createIconMenu(QMenu *pmenu)
{
    // Configuration of the tray icon (or dock icon) icon menu
    pmenu->addAction(toggleHideAction);
    pmenu->addSeparator();
    if (enableWallet) {
        pmenu->addAction(sendCoinsMenuAction);
        pmenu->addAction(coinJoinCoinsMenuAction);
        pmenu->addAction(receiveCoinsMenuAction);
        pmenu->addSeparator();
        pmenu->addAction(signMessageAction);
        pmenu->addAction(verifyMessageAction);
        pmenu->addSeparator();
        pmenu->addAction(openRPCConsoleAction);
        pmenu->addAction(openRepairAction);
        pmenu->addAction(showBackupsAction);
    }
    pmenu->addAction(optionsAction);
    pmenu->addAction(openInfoAction);
    pmenu->addAction(openGraphAction);
    pmenu->addAction(openPeersAction);
    pmenu->addSeparator();
    pmenu->addAction(openConfEditorAction);
#ifndef Q_OS_MAC // This is built-in on Mac
    pmenu->addSeparator();
    pmenu->addAction(quitAction);
#endif
}
```

Option 2) Keeping ordering
```
void BitcoinGUI::createIconMenu(QMenu *pmenu)
{
    // Configuration of the tray icon (or dock icon) icon menu
    pmenu->addAction(toggleHideAction);
    pmenu->addSeparator();
    if (enableWallet) {
        pmenu->addAction(sendCoinsMenuAction);
        pmenu->addAction(coinJoinCoinsMenuAction);
        pmenu->addAction(receiveCoinsMenuAction);
        pmenu->addSeparator();
        pmenu->addAction(signMessageAction);
        pmenu->addAction(verifyMessageAction);
        pmenu->addSeparator();
        pmenu->addAction(openRPCConsoleAction);
        pmenu->addAction(openRepairAction);
    }
    pmenu->addAction(optionsAction);
    pmenu->addAction(openInfoAction);
    pmenu->addAction(openGraphAction);
    pmenu->addAction(openPeersAction);
    pmenu->addSeparator();
    pmenu->addAction(openConfEditorAction);
    if (enableWallet) {
        pmenu->addAction(showBackupsAction);
    }
#ifndef Q_OS_MAC // This is built-in on Mac
    pmenu->addSeparator();
    pmenu->addAction(quitAction);
#endif
}
```

Reply:
Option 1 looks fine to me :)

Reply:
Ok, so I played with it a bit and I don't like Option 1 🙈 The tray menu looks weird/broken with this ordering. Also, `openRPCConsoleAction` was moved back to its current place in #15023. Let's backport it, move dash specific items back too and wrap them in place instead i.e. should look like that after all the fixes:
```
void BitcoinGUI::createIconMenu(QMenu *pmenu)
{
    // Configuration of the tray icon (or dock icon) icon menu
    pmenu->addAction(toggleHideAction);
    pmenu->addSeparator();
    if (enableWallet) {
        pmenu->addAction(sendCoinsMenuAction);
        pmenu->addAction(coinJoinCoinsMenuAction);
        pmenu->addAction(receiveCoinsMenuAction);
        pmenu->addSeparator();
        pmenu->addAction(signMessageAction);
        pmenu->addAction(verifyMessageAction);
        pmenu->addSeparator();
    }
    pmenu->addAction(optionsAction);
    pmenu->addAction(openInfoAction);
    pmenu->addAction(openRPCConsoleAction);
    pmenu->addAction(openGraphAction);
    pmenu->addAction(openPeersAction);
    if (enableWallet) {
        pmenu->addAction(openRepairAction);
    }
    pmenu->addSeparator();
    pmenu->addAction(openConfEditorAction);
    if (enableWallet) {
        pmenu->addAction(showBackupsAction);
    }
#ifndef Q_OS_MAC // This is built-in on Mac
    pmenu->addSeparator();
    pmenu->addAction(quitAction);
#endif
}
```

Reply:
Sounds good! backported 15023 as well and fixed the ordering as suggested.

I also reordered the commits to keep both "tray icon menu" commits close to each other